### PR TITLE
charts example

### PR DESCRIPTION
**Pull request by prakhargupta1**
Preview:

https://github.com/mui/mui-toolpad/assets/92228082/2ade96d4-13cf-493b-9925-e1fc1590481c

### CODE REVIEW

**Comment by prakhargupta1** [comment]
@apedroferreira In the 4th chart, I see the values when I hover over the chart, but nothing visually shows on the chart area.

**Comment by apedroferreira** [comment]
> @apedroferreira In the 4th chart, I see the values when I hover over the chart, but nothing visually shows on the chart area.

For scatter charts the data has to be numbers in both axes, I think that's a limitation that x-charts imposes... I can add an error message maybe to improve UX though.

File: examples/charts/toolpad/pages/page/page.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    - component: Chart
+      name: chart3
+      props:
+        data:
+          - label: No. of movies released per year
+            kind: scatter
+            data:
+              $$jsExpression: |
+                query1.data.map((movie) => ({
+                  year: movie.year,
```
Comment (apedroferreira) [inline examples/charts/toolpad/pages/page/page.yml:74]: For the scatter chart to show in this case:
```suggestion
                  year: Number(movie.year),
```

File: examples/charts/toolpad/pages/page/page.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                  count_id: query1.data.reduce((acc, currentMovie) => {
+                    return currentMovie.year === movie.year ? acc + 1 : acc
+                  }, 0),
+                }))
+            color: '#4caf50'
+            xKey: year
+            yKey: count_id
+  queries:
+    - name: query
```
Comment (apedroferreira) [inline examples/charts/toolpad/pages/page/page.yml:83]: There's 2 queries that are the same here. I guess we only need one.

**Review by apedroferreira — CHANGES_REQUESTED** [review]
Looks good, added just these suggestions.
Let me know if you have any more feedback about charts.

**Review by apedroferreira — APPROVED** [review]

**Pull request merged**